### ISSUE TITLE
changing date format issue options="{format: 'MM-DD-YYYY'}

### ISSUE THREAD

**Issue by RDhore** [fragaria/angular-daterangepicker#21]
I tried using the way below:
<input date-range-picker class="form-control date-picker" type="text" ng-model="vm.date" min="vm.min" max="vm.max" options="{format: 'MM-DD-YYYY'}"/>

In controller
$scope.vm = {
        date: {startDate: moment('01-01-2014', 'MM-DD-YYYY'), endDate: moment('01-31-2014', 'MM-DD-YYYY')},
        categories: '',
        outputCategories: [],
        min: moment('01-01-2014', 'MM-DD-YYYY'),
        max: moment('01-31-2014', 'MM-DD-YYYY')
    };

On model popup opened

$modalInstance.opened.then(function() {        
        
        var monthYearArray = monthYear.split(' ');
        var monthStart = moment().year(monthYearArray[1]).month(monthYearArray[0]).date(1).format('MM-DD-YYYY');
        var monthEnd = moment().year(monthYearArray[1]).month(monthYearArray[0]).date(31).format('MM-DD-YYYY');
        $log.log('monthYearArray', monthStart, monthEnd);
        
        $scope.vm.date.startDate = moment(monthStart, 'MM-DD-YYYY');
        $scope.vm.date.endDate = moment(monthEnd, 'MM-DD-YYYY');   
        $scope.vm.min = moment(monthStart, 'MM-DD-YYYY');
        $scope.vm.max = moment(monthEnd, 'MM-DD-YYYY');   
    });

It says invalid date

**Comment by HarelM** [comment]
I'm not sure this is relevant anymore but I had the same issue - its a problem of race condition of some sort. I didn't understand the code enough in order to be able to fix it, but I found a workaround:
I changed the date format in the directive js file. this worked for me since I'm not presenting different date format in different places in my app.
in angular-daterangepicker.js at the start of the file replace the code with this one:
<code>
picker.value('dateRangePickerConfig', {
    separator: ' - ',
    format: 'MM-DD-YYYY'
  });
</code>
Harel M.
